app/scripts/lib/strings.js

```diff
@@ -40,31 +44,62 @@ define(function (require, exports, module) {
    * the context
    * @method interpolate
    * @param {String} string
-   * @param {String} context
+   * @param {Object|Array} [context] - defaults to []
    * @returns {String}
    */
-  function interpolate(string, context) {
-    if (! context) {
-      context = [];
-    }
-
-    var interpolated = string.replace(/\%s/g, function (match) {
+  function interpolate(string, context = []) {
+    return string.replace(UNNAMED_VARIABLE_REGEXP, (match) => {
       // boot out non arrays and arrays with not enough items.
       if (! (context.shift && context.length > 0)) {
         return match;
       }
       return context.shift();
-    });
-
-    interpolated = interpolated.replace(/\%\(([a-zA-Z]+)\)s/g, function (match, name) {
+    }).replace(NAMED_VARIABLE_REGEXP, (match, name) => {
       return name in context ? context[name] : match;
     });
+  }
+
+  /**
+   * Check whether the string contains HTML. Very very loose interpretation of
+   * what HTML means. If it contains an HTML character code-like thing, it's
+   * considered HTML. If it contains <> with *anything* in the middle, it's
+   * considered HTML.
+   *
+   * @param {String} string
+   * @returns {Boolean} true if string has HTML, false otw.
+   */
+  function hasHTML(string) {
+    if (HTML_CHAR_CODE_REGEXP.test(string)) {
+      return true;
+    } else if (HTML_TAG_REGEXP.test(string)) {
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * Check whether all interpolation variables have an `escaped` prefix.
+   * This is a gentle reminder to developers that they must properly escape
+   * variables because the string is written to the DOM w/o HTML escaping.
+   *
+   * @param {String} string
+   * @returns {Boolean} true if has unsafe variables, false otw.
+   */
+  function hasUnsafeVariables(string) {
+    if (UNNAMED_VARIABLE_REGEXP.test(string)) {
```

Comment:
This could use `||` instead:

```js
return UNNAMED_VARIABLE_REGEXP.test(string) || UNSAFE_VARIABLE_REGEXP.text(string);
```

If you feel the `if`s read more easily, ignore me.